runtime/precompiles/src/StakingInterface.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// @nanocryk I've copied over the doc comments from the pallet. Is that useful to solidity devs?
+// Like specifically, will their IDEs show those docs or something? Otherwise, not worth copying.
+
+pragma solidity >=0.8.0;
+
+/// The interface through which solidity contracts will interact with Parachain Staking
+/// We follow this same interface including four-byte function selectors, in the precompile that
+/// wraps the pallet
+interface ParachainStaking {
+    // First some simple accessors
+
+    /// Check whether the specified address is currently a staking nominator
+    function is_nominator(address nominator) external view returns (bool);
+
+    /// Check whether the specified addess is currently a collator candidate
+    function is_candidate(address collator) external view returns (bool);
+
+    // Now the dispatchables
+
+    /// Join the set of collator candidates
+    function join_candidates(uint256 amount) external;
```

Comment:
It might be nice to document what this amount is (is it 10^-18 units?)

Reply:
It looks like it's used below as well, so maybe a comment at the top of the contract...

Reply:
A better variable name would go a long way. maybe `bond_amount`. Good news the names don't affect the signatures so we wouldn't have to regenerate them. Agree about the units. Luckily solidity makes this nice with keywords like `ether`, `wei`, etc.